test/errors/parsing/assertOnGpu.2.good

```diff
@@ -0,0 +1,21 @@
+─── error in assertOnGpu.chpl:2 [InvalidGpuAssertion] ───
+  Statement does not support the @assertOnGpu attribute.
+  The affected statement is here:
+      |
+    2 | record myRecord {
+      |
+
+─── error in assertOnGpu.chpl:4 [InvalidGpuAssertion] ───
+  Statement does not support the @assertOnGpu attribute.
+  The affected statement is here:
+      |
+    4 |     var A = foreach i in 1..100 do i;
+      |
```

Comment:
https://github.com/chapel-lang/chapel/pull/24366/files/31f0baeba311b897a6cfb5281d54b5b1b2695aff..c9ee43614c871f406764a54fa4d3c444f3a14a40#diff-65a977858b3edee735c3ecc9eb0223f27c729f092056ea41c7a65de1aea05e40 this suggests that it is supported, but maybe the problem is that this is a field initializer?

Reply:
(I think you mentioned this to me on Slack) Just in case it is easy: can we make the error message make it clear that this statement is not supported because it is a field initialization?